### PR TITLE
CI fixes

### PR DESCRIPTION
**Pull request by yosefe**
## Why
1. Fix assertion in rcx/test_ucp_tag_mem_type.reuse_buffers_mrail/15. The receive buffer could have been registered as part of tag offload and moved from recv iterator to rndv send request iterator.
2. Fix the condition that enable testing protov2 specific features: need to check UCP configuration rather than test variant, since protov2 can be disabled by env var UCX_PROTO_ENABLE=n

cc @shasson5

### CODE REVIEW

File: test/gtest/ucp/test_ucp_rma.cc [hunk 1]
```diff
@@ -182,7 +182,7 @@ class test_ucp_rma : public test_ucp_memheap {
     }
 
     bool disable_proto() {
-        return !m_ucp_config->ctx.proto_enable;
+        return !is_proto_enabled();
```
Comment (rakhmets) [inline test/gtest/ucp/test_ucp_rma.cc:185]: I guess it's a bug here (from this PR https://github.com/openucx/ucx/pull/9084).
This method should check test variant value, and the config value should be set based on return value.
Thus, it should be:
 ```C
 bool disable_proto() const
{
        return get_variant_value() & DISABLE_PROTO;
}
```

And then `!is_proto_enabled()` should be used in this file at these lines: 295 and 314.

File: test/gtest/ucp/test_ucp_mmap.cc [hunk 2]
```diff
@@ -424,7 +424,7 @@ void test_ucp_mmap::test_rkey_proto(ucp_mem_h memh)
     ASSERT_UCS_OK(status);
 
     /* Check rkey configuration */
-    if (!disable_proto() && m_ucp_config->ctx.proto_enable) {
+    if (!disable_proto() && is_proto_enabled()) {
```
Comment (rakhmets) [inline test/gtest/ucp/test_ucp_mmap.cc:427]: Guess that should be enough:
```suggestion
    if (is_proto_enabled()) {
```

File: test/gtest/ucp/test_ucp_atomic.cc [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 extern "C" {
 #include <ucp/core/ucp_types.h> /* for atomic mode */
+#include <ucp/core/ucp_context.h>
```
Comment (rakhmets) [inline test/gtest/ucp/test_ucp_atomic.cc:12]: This line can be removed.

File: test/gtest/ucp/test_ucp_rma.cc (hunk 1 above)
Comment (brminich) [reply test/gtest/ucp/test_ucp_rma.cc:185]: config maybe set externally (e.g. in proto_disable roce tests in CI)

Comment (rakhmets) [reply test/gtest/ucp/test_ucp_rma.cc:185]: I see.
But we don't use `DISABLE_PROTO` variant value in these tests.

File: test/gtest/ucp/test_ucp_rma.cc [hunk 4]
```diff
@@ -182,7 +182,7 @@ class test_ucp_rma : public test_ucp_memheap {
     }
 
     bool disable_proto() {
-        return !m_ucp_config->ctx.proto_enable;
+        return get_variant_value() & DISABLE_PROTO;
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_rma.cc:185]: seems it is only used in `init` method now.
maybe it's better for check for `get_variant_value() & DISABLE_PROTO` there directly?

Comment (brminich) [reply test/gtest/ucp/test_ucp_rma.cc:184]: @yosefe, do you think it's worth to cnosider?
seems such method is removed in other places

Comment (yosefe) [reply test/gtest/ucp/test_ucp_rma.cc]: done

**Review by rakhmets — DISMISSED** [review]

**Pull request merged**